doc: services: device_mgmt: mcumgr: Remove references to go tool

nordicjm opened the pull request:
The go tool documentation is very old, and the tool is not recommended for use, therefore remove all documentation relating to it to prevent people from using it

nordicjm committed:
doc: services: device_mgmt: mcumgr: Remove references to go tool

The go tool documentation is very old, and the tool is not
recommended for use, therefore remove all documentation relating
to it to prevent people from using it

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
samples: mgmt: mcumgr: smp_svr: Remove documentation for go tool

Removes references to old tool documentation and refers users to
the list of supported applications instead

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

kartben commented on samples/subsys/mgmt/mcumgr/smp_svr/README.rst:
```diff
@@ -287,80 +237,47 @@ For clarity, here is DTS label to slot to ``<image>`` translation table:
-   sudo mcumgr <connection string> reset
-
-Upon reset MCUboot will swap slot-0 and slot-1.
+We can reset the device remotely to observe (use the console output) how MCUboot swaps the images,
+check the documentation in the tool of your choice. Upon reset MCUboot will swap slot-0 and
```
non blocking but the mixed use of "you" vs. "we" across this page should probably be improved at some point

nordicjm replied:
Made some changes

kartben reviewed (changes requested):
> The go tool documentation is very old, and the tool is not recommended for use

My only request for change is that I'd like to see an entry for this in the release notes to make it extra clear to folks that might still be using this tool that they should probably look for other options. I wouldn't go as far as adding something to the migration guide too, but if you feel there's value in being even more proactive in giving folks recommendations for a replacement, go ahead.

nordicjm committed:
doc: release: 3.7: Add note on MCUmgr go tool instruction removal

Adds a note that these instructions have been removed and list
the available replacement tools

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

vrahane commented:
mcumgr CLI tool has matured quite a bit ever since and go libraries can be updated as needed by devs. We still use it and works great for us, we have updated serial/coap/ble libraries internally. Though, I do understand if you guys are moving away from Go. Not a very widely used language in the embedded world.
